### PR TITLE
Flannel fixes and enhancements

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6797)

### CODE REVIEW

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-me-please

File: pkg/endpoint/endpoint.go [hunk 1]
```diff
@@ -1171,7 +1171,7 @@ func FilterEPDir(dirFiles []os.FileInfo) []string {
 	for _, file := range dirFiles {
 		if file.IsDir() {
 			_, err := strconv.ParseUint(file.Name(), 10, 16)
-			if err == nil || strings.HasSuffix(file.Name(), "_next_fail") {
+			if err == nil || strings.HasSuffix(file.Name(), "_next") || strings.HasSuffix(file.Name(), "_next_fail") {
```
Comment (tgraf) [inline pkg/endpoint/endpoint.go]: Please spin this in a separate PR

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-flannel

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Flannel/11/

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-flannel

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-flannel

**Review by eloycoto — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-flannel

**Review by ianvernon — APPROVED** [review]
